rails-commenting.rb

```diff
@@ -6,25 +6,25 @@
 
 # FILE: app/controller/blog_posts_controller.rb
 
-# ---1)
+# ---1) This line establashes the class of 'BlogPostsController' which serves as the controller, or what I like to think of as the switchboard, of a Rails application. The methods are the switches and when a switch is "flipped" it tells the application what to do. The BlogPostsController class is inheriting its properties from the parent 'ApplicationController'.
 class BlogPostsController < ApplicationController
   def index
-    # ---2)
+    # ---2) Posts is an instance variable, identifiable by the '@' symbol preceeding the word. This makes 'posts' accessible from outside of the method it is instantiated in. The value assigned to @posts is the entirety of the model, or database, called BlogPost.
     @posts = BlogPost.all
   end
 
-  # ---3)
+  # ---3) This is the beginning of a method called 'show' whose job in the controller is to single out an instance from the BlogPost model. Specifically, this line is where the method is being "defined".
   def show
     @post = BlogPost.find(params[:id])
   end
 
-  # ---4)
+  # ---4) The method being defined here is 'new', whose purpose is to communicate the creation of a new entry into the database that it is connected to. 'New' readys the database for the incoming entry, it does not take in the information that will be associated with the new entry. 
```

Comment:
Nice!  This one is often tricky to describe and you did a great job in explaining it.